### PR TITLE
[SYCLomatic] Remove warning for clock function in host functions

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -6894,11 +6894,14 @@ void FunctionCallRule::runRule(const MatchFinder::MatchResult &Result) {
     emplaceTransformation(EA.getReplacement());
     EA.applyAllSubExprRepl();
   } else if (FuncName == "clock" || FuncName == "clock64") {
-    report(CE->getBeginLoc(), Diagnostics::API_NOT_MIGRATED_SYCL_UNDEF, false,
-           FuncName);
-    // Add '#include <time.h>' directive to the file only once
-    auto Loc = CE->getBeginLoc();
-    DpctGlobalInfo::getInstance().insertHeader(Loc, HT_Time);
+    if (CE->getCalleeDecl()->hasAttr<CUDAGlobalAttr>() ||
```
Comment (ziranzha) [inline clang/lib/DPCT/ASTTraversal.cpp]: `CallExpr::getCalleeDecl()` may return `nullptr`, need check before deref it.
BTW, use `getDirectCallee()` may be better.

Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/ASTTraversal.cpp]: Replaced with getDirectCallee(). 

Nullptr is checked here : 

https://github.com/oneapi-src/SYCLomatic/blob/cde21e5b3e5ae449e8c5266a8f3b71cd01161847/clang/lib/DPCT/ASTTraversal.cpp#L6726

**Review by ziranzha — APPROVED** [review]

**Pull request merged**